### PR TITLE
feat(RHINENG-17791): Add "ungrouped" to Group serialization

### PR DESCRIPTION
**Pull request by kruai**
# Overview

This PR is being created to address [RHINENG-17791](https://issues.redhat.com/browse/RHINENG-17791).
Adds the "ungrouped" field to Groups serialization. This means that the "ungrouped" field is now present on the API response for Group, as well as Host.groups. This will enable the UI to make decisions that require this information, such as assigning a Host in the "ungrouped" Group to another Group.

Here is the docs update MR for this change: https://gitlab.cee.redhat.com/consoledot/consoledot.pages.redhat.com/-/merge_requests/306

## PR Checklist

- [x] Keep PR title short, ideally under 72 characters
- [ ] Descriptive comments provided in complex code blocks
- [ ] Include raw query examples in the PR description, if adding/modifying SQL query
- [ ] Tests: validate optimal/expected output
- [ ] Tests: validate exceptions and failure scenarios
- [ ] Tests: edge cases
- [ ] Recovers or fails gracefully during potential resource outages (e.g. DB, Kafka)
- [ ] Uses [type hinting](https://docs.python.org/3/library/typing.html), if convenient
- [x] Documentation, if this PR changes the way other services interact with host inventory
- [x] Links to related PRs

## Secure Coding Practices Documentation Reference

You can find documentation on this checklist [here](https://github.com/RedHatInsights/secure-coding-checklist).

## Secure Coding Checklist

- [ ] Input Validation
- [ ] Output Encoding
- [ ] Authentication and Password Management
- [ ] Session Management
- [ ] Access Control
- [ ] Cryptographic Practices
- [ ] Error Handling and Logging
- [ ] Data Protection
- [ ] Communication Security
- [ ] System Configuration
- [ ] Database Security
- [ ] File Management
- [ ] Memory Management
- [ ] General Coding Practices

## Summary by Sourcery

Add "ungrouped" field to Group serialization to support identifying ungrouped hosts in the API response

New Features:
- Add 'ungrouped' boolean field to Group serialization to indicate if a group represents ungrouped hosts

Enhancements:
- Updated serialization method to include 'ungrouped' flag
- Updated API specification to document the new 'ungrouped' field

Tests:
- Added test cases to verify ungrouped hosts query and group serialization
- Expanded test coverage for group and host queries with ungrouped groups

### CODE REVIEW

File: tests/test_api_hosts_get.py [hunk 1]
```diff
@@ -746,6 +749,20 @@ def test_query_ungrouped_hosts(db_create_group_with_hosts, mq_create_three_speci
...
+    ungrouped_group_id = db_create_group_with_hosts("ungrouped", 3, True).id
+    url = build_hosts_url(query="?group_name=ungrouped")
+
+    response_status, response_data = api_get(url)
+
+    assert response_status == 200
+    for result in response_data["results"]:
+        assert (group_result := result["groups"][0])["name"] == "ungrouped"
+        assert group_result["id"] == str(ungrouped_group_id)
+        assert group_result["ungrouped"] is True
```
[the remark this answers is not in the document]
Comment (kruai) [reply tests/test_api_hosts_get.py]: Removed.

File: tests/test_api_hosts_get.py [hunk 2]
```diff
@@ -746,6 +749,20 @@ def test_query_ungrouped_hosts(db_create_group_with_hosts, mq_create_three_speci
     assert_host_lists_equal(build_expected_host_list(ungrouped_hosts), response_data["results"])
```
[the remark this answers is not in the document]
Comment (kruai) [reply tests/test_api_hosts_get.py:749]: Doesn't make sense for our implementation. If the host is not in a group, it will not have an "ungrouped" group in its response.

**Comment by kruai** [comment]
> I don't see reasoning behind adding the field in the PR nor the ticket. I feel like it might be just a system flag, that a user doesn't care about 🤔 Just for my education, do we actually need this field exposed? Why?

@ezr-ondrej I attempted to briefly cover it briefly in my PR description, but to go into more detail, it's needed for the UI to distinguish between "ungrouped" groups and regular ones. Users are not allowed to add a host to a group if the host is already in a group, and the UI enforces this by checking whether host.groups is empty (thus ungrouped). However, after the data migration, all hosts will be in a group: hosts that are not currently in a group get added to an "ungrouped" group. With this new requirement, the UI can no longer rely on whether host.groups is empty to determine whether to enable certain functionality. So, this adds an "ungrouped" boolean to host.groups so that the UI can make these decisions.

**Comment by ezr-ondrej** [comment]
Gotcha, that makes sense, I guess it would be much harder to do it any other way 👍
Sometimes I wonder if we should have API for UI and have two APIs (one for customers and second for UI).
